CLDC-2001 Bulk upload stacked errors

asmega opened the pull request:
# Context

- https://digital.dclg.gov.uk/jira/browse/CLDC-2001
- There are times that if there is an field is incorrect several errors are added a result
- We want to reduce these so there is less noise and users can focus on the issues that they can action to resolve problematic data

# Changes

- Only include existing log errors if the field does not already have errors on it. this allows the CSV errors to supersede the existing log errors 
- Shim in additional errors for ages for bulk upload to prevent quirky existing errors from bubbling up 

# Notes for review

- We're now entering more into the territory where existing log validations are simply insufficient and do not work with the desired effect for bulk upload although are applicable for the web flow
- As a result we have to shim in desired validations at CSV level and only include existing validation errors if they shimmed rules are not triggered

asmega committed:
limit setup errors if errors already present

asmega committed:
bulk upload beter handles age validations

- limit the number of errors on a field. if the field already has an
  existing error it does not add further errors to the field
- shim in extra validation for ages which take precedent over existing
  log level validations which are lacking

kosiakkatrina commented on app/services/bulk_upload/lettings/row_parser.rb:
```diff
@@ -387,7 +408,11 @@ def validate_nulls
 
-      fields.each { |field| errors.add(field, I18n.t("validations.not_answered", question: question.check_answer_label&.downcase)) }
+      fields.each do |field|
+        unless errors.any? { |e| fields.include?(e.attribute) }
```
What is this doing?

asmega replied:
- yep so before if there was an error it was always get added to the relevant fields
- now if any one of the relevant fields has one error on it we no longer keeping adding errors to that grouping

eg `startdate` consists of `field_dd`, `field_mm`, `field_yy` from the mapping
if there is an error that already on `field_dd` the code will no longer surface any additional errors on those fields. this should prevent the user from being overwhelmed by too many errors

natdeanlewissoftwire commented on app/services/bulk_upload/lettings/row_parser.rb:
```diff
@@ -337,7 +356,7 @@ def validate_leaving_reason_for_renewal
 
   def validate_relevant_collection_window
-    return unless start_date && bulk_upload.form
+    return if start_date.blank? || bulk_upload.form.blank?
```
I'm with you on not using unless UNLESS it's actually useful, which I agree it isn't here

natdeanlewissoftwire reviewed (approved):
Looks good, my domain knowledge on bulk upload isn't as strong so I'm deferring to yours as to why these validations are necessary to be separate, why we're allowing R for age etc.

asmega committed:
setup errors always added

- no longer observes if the grouping already has errors
- as we have fine grain control in this class of how errors should be

asmega committed:
fix missing pass through for renewal
